azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -365,33 +363,19 @@ private void createLogger() {
             + " for job " + this.jobId, e);
       }
 
-      if (this.props.getBoolean(Constants.JobProperties.AZKABAN_JOB_LOGGING_KAFKA_ENABLE, false)) {
-        // Only attempt appender construction if required properties are present
-        if (this.azkabanProps
-            .containsKey(Constants.ConfigurationKeys.AZKABAN_SERVER_LOGGING_KAFKA_BROKERLIST)
-            && this.azkabanProps
-            .containsKey(Constants.ConfigurationKeys.AZKABAN_SERVER_LOGGING_KAFKA_TOPIC)) {
-          try {
-            attachKafkaAppender(createKafkaAppender());
-          } catch (final Exception e) {
-            removeAppender(this.kafkaAppender);
-            this.flowLogger.error("Failed to create Kafka appender for job " + this.jobId, e);
-          }
-        } else {
-          this.flowLogger.info(
-              "Kafka appender not created as brokerlist or topic not provided by executor server");
+      if (this.azkabanProps.getBoolean(Constants.ConfigurationKeys.AZKABAN_LOGGING_KAFKA_ENABLED, false)) {
+        this.kafkaLog4jAppender =
+            KafkaLog4jUtils.getAzkabanJobKafkaLog4jAppender(this.azkabanProps,
+                String.valueOf(this.executionId),  this.node.getParentFlow().getFlowId(),
```

Comment:
Is 'this.node.getParentFlow().getFlowId()' valid for nested flow?

Reply:
It is an excellent question! Thank you so much for bringing this up!

I find when we upload or query logs from mysql, we are using this.node.getNestedId() only for the query key. 'this.node.getParentFlow().getFlowId()' is not enough for nested flows but this.node.getNestedId() is the best representative for a node.

Also, I changed the getAzkabanJobKafkaLog4jAppender interface accordingly.